### DISCUSSION TITLE
Endstops

### DISCUSSION THREAD

**Discussion by Festivejelly** [kachurovskiy/nanoels#174] [Ideas]
I think having endstops on the X would be quite useful. One could first take a light cut, measure and zero the X axis based on the diameter. But then one could "home" the X axis and when it hits the microswitch we could take a reading of the X value, then this could be recorded in the config. 

What this means is when you start you machine you can simple home the X and you should have a pretty damn repeatable centre line. This would obv only work for the tool you calibrated it with, but other tools offsets could be calibrated too as per my PR idea for tool changes.

With high quality switches you should get some good repeatability. Probably enough for the hobbyist. Or maybe even use inductive switches like they do on CNC machines.

**Comment by vegetate7** [comment]
I do not think there are free IO lines lasts on ESP32 module. But this is nice idea. Not limit switches only. Could be also mechanical switches for axis enable/disable, change direction, etc...

**Reply by Festivejelly** [reply]
I guess one could re use an existing io port? Im not using all of them.

**Comment by kachurovskiy** [comment]
The great power of open source - anyone can make changes they'd like to see and share them :)

Homing on hobby machines is pretty hard to make useful. Homing Z seems of little benefit since the stock is rarely precisely aligned on Z - it's just inserted into the chuck manually. Homing X is hard because the result of a cut depends on how the tool is set today, how far away from the chuck we're cutting, stock material, deflection, etc. Investing time into this doesn't seem favorable to me from the risk/reward perspective.

**Reply by Festivejelly** [reply]
I think you’re right, this is probably only going to be beneficial on very precise machines with a turret that is super repeatable.
